libraries/chain/fork_database.cpp

```diff
@@ -192,10 +192,11 @@ namespace eosio::chain {
 
    template<class BSP>
    void fork_database_impl<BSP>::reset_root_impl( const bsp_t& root_bsp ) {
-      index.clear();
       assert(root_bsp);
       root = root_bsp;
       root->set_valid(true);
+      pending_savanna_lib_id = block_id_type{};
+      index.clear();
    }
 
```

Comment:
The reset of `pending_savanna_lib_id` is needed as starting in irreversible mode can reset the forkdb root to head. If the node is started again then it would try and advance LIB and segfault because the fork db didn't have anything in the index which means `forkdb.head()` returns nullptr. I attempted to reproduce this in 1.0.x, but couldn't find a way to trigger it. Not sure if we should backport it to 1.0.x or not.

Reply:
Maybe not backport for now as it is a rare case and 1.1.x is not far away.